suite-native/firmware/src/hooks/useFirmwareAnalytics.ts

```diff
@@ -0,0 +1,112 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+import { DeviceModelInternal, FirmwareType } from '@trezor/connect';
+import {
+    analytics,
+    EventType,
+    FirmwareUpdatePayload,
+    FirmwareUpdateStartType,
+} from '@suite-native/analytics';
+import { TrezorDevice } from '@suite-common/suite-types';
+import { getFirmwareVersion, getBootloaderVersion } from '@trezor/device-utils';
+
+export const useFirmwareAnalytics = ({
+    device,
+    targetFirmwareType,
+}: {
+    device?: TrezorDevice;
+    targetFirmwareType: FirmwareType;
+}) => {
+    const prepareAnalyticsPayload = useCallback(() => {
+        return {
+            model: device?.features?.internal_model ?? DeviceModelInternal.UNKNOWN,
+            fromBootloaderVersion: getBootloaderVersion(device),
```

Comment:
This will most likely not work or always return only empty string, because `getBootloaderVersion` util works only if the device is in bootloader mode. I think we don't have a way how to read the bootloader version in normal mode. But we can assume the BL version from FW version anyway, so this is probably redundant.